editor/src/components/editor/store/remix-derived-data.tsx

```diff
@@ -0,0 +1,109 @@
+import type {
+  UNSAFE_FutureConfig as FutureConfig,
+  UNSAFE_AssetsManifest as AssetsManifest,
+} from '@remix-run/react'
+import type { MutableUtopiaCtxRefData } from '../../canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-contexts'
+import type { MapLike } from 'typescript'
+import type { ComponentRendererComponent } from '../../canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-component-renderer'
+import type { DataRouteObject } from 'react-router'
+import type { ProjectContentTreeRoot } from '../../assets'
+import type {
+  RouteIdsToModuleCreators,
+  RouteModulesWithRelativePaths,
+} from '../../canvas/remix/remix-utils'
+import {
+  DefaultFutureConfig,
+  createAssetsManifest,
+  createRouteManifestFromProjectContents,
+  getRoutesAndModulesFromManifest,
+} from '../../canvas/remix/remix-utils'
+import type { CurriedUtopiaRequireFn, CurriedResolveFn } from '../../custom-code/code-file'
+import type { ElementPath } from '../../../core/shared/project-file-types'
+import type { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import type { AllElementProps, CanvasBase64Blobs } from './editor-state'
+import { memoize } from '../../../core/shared/memoize'
+
+export interface RemixStaticRoutingTable {
+  [rootElementUid: string]: string /* file path */
+}
+
+export interface RemixDerivedData {
+  futureConfig: FutureConfig
+  assetsManifest: AssetsManifest
+  routeModuleCreators: RouteIdsToModuleCreators
+  routeModulesToRelativePaths: RouteModulesWithRelativePaths
+  routes: Array<DataRouteObject>
+  routingTable: RemixStaticRoutingTable
+}
+
+const CreateRemixDerivedDataRefs: {
+  mutableContext: { current: MutableUtopiaCtxRefData }
+  topLevelComponentRendererComponents: { current: MapLike<MapLike<ComponentRendererComponent>> }
+  resolvedFiles: { current: MapLike<Array<string>> }
+  resolvedFileNames: { current: Array<string> }
+} = {
+  mutableContext: { current: {} },
+  topLevelComponentRendererComponents: { current: {} },
+  resolvedFiles: { current: {} },
+  resolvedFileNames: { current: [] },
+}
```

Comment:
Do we still need this?

Reply:
Cleaning this up would happen in a separate PR, I removed the unused props in https://github.com/concrete-utopia/utopia/pull/4162/commits/fbac022f2c91c1b9dcd98b423f0cc6712297d3a5